lib/src/player.rs

```diff
@@ -62,12 +62,27 @@ pub async fn get_highest_id_with_sets_between(start_id: i32, end_id: i32) -> Res
 pub async fn get_all_like(tag: &str) -> Result<Vec<Player>> {
     let processed_tag = tag.replace(' ', "%");
     // ^^^ transform spaces into wildcards to make search more inclusive
-
     let mut db_connection = smithe_database::connect().await?;
-    let matching_players: Vec<Player> = players
-        .filter(gamer_tag.ilike(format!("%{}%", processed_tag))) // case-insensitive like
-        .get_results::<Player>(&mut db_connection)
-        .await?;
+
+    let mut matching_players: Vec<Player> = sql_query(format!(
+        "SELECT * FROM players WHERE gamer_tag ILIKE '%{}%' ORDER BY LENGTH(gamer_tag) ASC, player_id ASC;",
+        processed_tag
+    ))
+    .get_results::<Player>(&mut db_connection)
+    .await?;
```

Comment:
As is, this is vulnerable to SQL injections - if using Diesel's constructs is too hard for this specific example, using a prepared statement is the best way forward.

It would look like this:

```suggestion
      let query = diesel::sql_query("SELECT * FROM players WHERE gamer_tag ILIKE $1 ORDER BY LENGTH(gamer_tag) ASC, player_id ASC")
      .bind::<Text, _>(format!("%{}%", processed_tag));

    let mut matching_players: Vec<Player> = query.get_results::<Player>(&mut db_connection).await?;
```

You can accept this suggestion, but you'll prob. have to import that `Text` struct in the `bind`, btw.

Also - If you are curious, you can see we have a SQL injection vulnerability by passing smt like: `dummy%'`. As is, we get a panic. W/ the prepared statement, we correctly show 'No Results'.